tests/AutoReview/ProjectCodeTest.php

```diff
@@ -208,34 +208,63 @@ public function testThatSrcClassesNotExposeProperties($className)
     }
 
     /**
-     * @param string $className
-     *
      * @dataProvider provideTestClassCases
+     *
+     * @param string $testClassName
      */
-    public function testThatTestClassesAreTraitOrAbstractOrFinal($className)
+    public function testThatTestClassesAreTraitOrAbstractOrFinal($testClassName)
     {
-        $rc = new \ReflectionClass($className);
+        $rc = new \ReflectionClass($testClassName);
 
         static::assertTrue(
             $rc->isTrait() || $rc->isAbstract() || $rc->isFinal(),
-            sprintf('Test class %s should be trait, abstract or final.', $className)
+            sprintf('Test class %s should be trait, abstract or final.', $testClassName)
         );
     }
 
     /**
-     * @param string $className
-     *
      * @dataProvider provideTestClassCases
+     *
+     * @param string $testClassName
      */
-    public function testThatTestClassesAreInternal($className)
+    public function testThatTestClassesAreInternal($testClassName)
     {
-        $rc = new \ReflectionClass($className);
+        $rc = new \ReflectionClass($testClassName);
         $doc = new DocBlock($rc->getDocComment());
 
         static::assertNotEmpty(
             $doc->getAnnotationsOfType('internal'),
-            sprintf('Test class %s should have internal annotation.', $className)
+            sprintf('Test class %s should have internal annotation.', $testClassName)
+        );
+    }
+
+    /**
+     * @dataProvider provideTestClassCases
+     *
+     * @param string $testClassName
+     */
+    public function testThatPublicMethodsAreCorrectlyNamed($testClassName)
+    {
+        $reflectionClass = new \ReflectionClass($testClassName);
+
+        $publicMethods = array_filter(
+            $reflectionClass->getMethods(\ReflectionMethod::IS_PUBLIC),
+            static function (\ReflectionMethod $reflectionMethod) use ($reflectionClass) {
+                return $reflectionMethod->getDeclaringClass()->getName() === $reflectionClass->getName();
+            }
         );
+
+        if ($publicMethods === []) {
+            $this->addToAssertionCount(1); // no methods to test, all good!
+        }
+
+        foreach ($publicMethods as $method) {
+            static::assertRegExp(
+                '/^(test|provide|setUpBeforeClass$|tearDownAfterClass$)/',
+                $method->getName(),
+                sprintf('Public method "%s::%s" is not properly named.', $reflectionClass->getName(), $method->getName())
```

Comment:
when i would run newly written tests and see this msg as an outcome in console, i would have no clue why given method is not "properly named"

Reply:
wouldn't you go then to the test and see the regex? can you [propose](https://thenextweb.com/dd/2018/10/16/github-launches-suggested-changes-to-make-it-easier-to-collaborate-within-pull-requests/) the message?